### PR TITLE
feat: describe EnumVal

### LINKED ISSUE #1: describeEnumVal
As a user of scl-diff I would like to have an object describing the SCL element `EnumVal`.

**Requirements:**
- return `EnumValDescription`
```ts
type EnumValDescription = {
    /** the attribute `ord` of the element `EnumVal` */
    ord?: number;
    /** the attribute `desc` of the element  `EnumVal` */
    desc?: string;
   /** the textCntent of the element `EnumVal` */
   content?: string;
}
```

### PR DESCRIPTION
**Pull request by JakobVogelsang**
Closes #1

### CODE REVIEW

File: describe/EnumVal.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export type EnumValDescription = {
+  /** the attribute `ord` of the element `EnumVal` */
+  ord?: number;
```
Comment (ca-d) [inline describe/EnumVal.ts:3]: Why does this have a question mark? Can it be undefined under some circumstances?

File: describe/EnumVal.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  /** the attribute `desc` of the element  `EnumVal` */
+  desc?: string;
+  /** the textContent of the element `EnumVal` */
+  content?: string;
```
Comment (ca-d) [inline describe/EnumVal.ts:7]: "content" sounds a bit vague to me. Do you think `val` might be a clearer name?

File: describe.spec.ts [hunk 3]
```diff
@@ -22,6 +27,7 @@ const privateElement = testScl.querySelector("Private")!;
 const sclElement = testScl.querySelector("SCL")!;
 const SomeNonSCLElement = testScl.querySelector("SomeNonSCLElement")!;
 const textElement = testScl.querySelector("Text")!;
+const enumValElement = testScl.querySelector("EnumVal")!;
 
 describe("Describe SCL elements function", () => {
   it("returns undefined with missing describe function", () =>
```
Comment (ca-d) [inline describe.spec.ts:33]: In terms of test names, I'd rather have them read like

```
describe(element) -> is undefined for non-namespaced non-SCL elements
describe(element) -> describes EnumVal elements
```

etc. What do you think about such a naming scheme?

File: describe/EnumVal.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  "application/xml"
+);
+
+const enumVal = scl.querySelector('EnumVal[ord="1"]')!;
+const enumValMissingContent = scl.querySelector('EnumVal[ord="2"]')!;
+const nonSenseOrd = scl.querySelector('EnumVal[ord="nonsense"]')!;
+
+describe("Description for SCL element EnumVal", () => {
+  it("return ord, desc attribute and innerText content ", () =>
+    expect(EnumVal(enumVal)).to.deep.equal({
```
Comment (ca-d) [inline describe/EnumVal.spec.ts:26]: These deep equal tests smell a bit like repeating implementation details. Maybe we want to guarantee all these implementation details as part of the describe API for certain use cases, but at least for hashing and diffing, the important thing to us would be that certain elements are described the same while others are described differently, if that makes sense.

So maybe tests like

```ts
expect(EnumVal(testVal)).to.deep.equal(EnumVal(equalVal));
expect(EnumVal(testVal)).to.not.deep.equal(EnumVal(differentVal));
```

and so on might suffice. What do you think about this? I'm torn.

**Review by ca-d — CHANGES_REQUESTED** [review]
I'd request you change the type definition if (and only if) `ord` can't be `undefined`, the other comments are naming suggestions and a question about general testing strategy that I'm unsure of myself.

File: describe/EnumVal.ts (hunk 1 above)
Comment (JakobVogelsang) [reply describe/EnumVal.ts:3]: Very good question @ca-d . I am really unsure about this myself. The reason is this: The attribute `ord` in `EnumVal` is required. The question though is how do we handle schema invalid files. XML is not enforcing the attribute to be set. I had several ideas here:

1. Return `undefined` for invalid `EnumVal` element. 
2. Return `NaN` for this specific case

Btw. that is a general question. How do we handle required attributes that incorrectly are missing.

File: describe/EnumVal.ts (hunk 2 above)
Comment (JakobVogelsang) [reply describe/EnumVal.ts:7]: Thank you yes!

File: describe.spec.ts (hunk 3 above)
Comment (JakobVogelsang) [reply describe.spec.ts:33]: To be honest with you, I do not understand your proposal, sorry. I am very open to discuss this further.

**Comment by JakobVogelsang** [comment]
Not applicable due to changed scope of the issue. See #1

**Pull request closed without merging**